python/lsst/daf/butler/core/quantum.py

```diff
@@ -21,49 +21,88 @@
+    ``runQuantum`` method, but they can also be used to represent discrete
+    units of work performed manually by human operators or other software
+    agents.
 
     Parameters
     ----------
-    task : `str` or `SuperTask`
-        Fully-qualified name of the SuperTask that executed this Quantum.
-    run : `Run`
+    taskName : `str`, optional
+        Fully-qualified name of the Task class that executed or will execute
+        this Quantum.  If not provided, ``taskClass`` must be.
+    taskClass : `type`, optional
+        The Task class that executed or will execute this Quantum.  If not
+        provided, ``taskName`` must be.  Overrides ``taskName`` if both are
+        provided.
+    dataId : `DataId`, optional
+        The dimension values that identify this `Quantum`.
+    run : `Run`, optional
         The Run this Quantum is a part of.
+    initInputs : collection of `DatasetRef`
+        Datasets that are needed to construct an instance of the Task.  May
+        be a flat iterable of `DatasetRef` instances or a mapping from
+        `DatasetType` to `DatasetRef`.
+    predictedInputs : `~collections.abc.Mapping`, optional
+        Inputs identified prior to execution, organized as a mapping from
+        `DatasetType` to a list of `DatasetRef`.  Must be a superset of
+        ``actualInputs``.
+    actualInputs : `~collections.abc.Mapping`, optional
+        Inputs actually used during execution, organized as a mapping from
+        `DatasetType` to a list of `DatasetRef`.  Must be a subset of
+        ``predictedInputs``.
+    outputs : `~collections.abc.Mapping`, optional
+        Outputs from executing this quantum of work, organized as a mapping
+        from `DatasetType` to a list of `DatasetRef`.
+    kwargs
+        Additional arguments are forwarded to the base `Execution` constructor.
     """
 
-    __slots__ = ("_task", "_run", "_predictedInputs", "_actualInputs", "_outputs")
-    __eq__ = slotValuesAreEqual
-
-    def __init__(self, task, run, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._task = task
+    __slots__ = ("_taskName", "_taskClass", "_dataId", "_run",
+                 "_initInputs", "_predictedInputs", "_actualInputs", "_outputs")
+
+    def __init__(self, *, taskName=None, taskClass=None, dataId=None, run=None,
+                 initInputs=None, predictedInputs=None, actualInputs=None, outputs=None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        if taskClass is not None:
+            taskName = f"{taskClass.__module__}.{taskClass.__name__}"
+        self._taskName = taskName
+        self._taskClass = taskClass
         self._run = run
-        self._predictedInputs = {}
-        self._actualInputs = {}
-        self._outputs = {}
+        self._dataId = dataId
+        if initInputs is None:
+            initInputs = {}
+        elif not hasattr(initInputs, "keys"):
+            initInputs = {ref.datasetType: ref for ref in initInputs}
```

Comment:
No, the `PipelineTask` declarations already assume all `initInput` and `initOutput` dataset types are scalar (and they have to be because they have empty data IDs).